chall02/ptuukkan.c

```diff
@@ -0,0 +1,45 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ptuukkan.c                                         :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: ptuukkan <ptuukkan@student.hive.fi>        +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2020/03/30 14:03:53 by ptuukkan          #+#    #+#             */
+/*   Updated: 2020/03/30 14:27:54 by ptuukkan         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+static char	ft_itoc(int i)
+{
+	if (i < 10)
+		return (i + '0');
+	return (i + 87);
+}
+
+char		*hv_rgb2hex(int r, int g, int b)
+{
+	static char	rgb[8];
+
+	if (r > 255)
+		r = 255;
+	else if (r < 0)
+		r = 0;
+	if (g > 255)
+		g = 255;
+	else if (g < 0)
+		g = 0;
+	if (b > 255)
+		b = 255;
+	else if (b < 0)
+		b = 0;
```

Comment:
it's nice you protected; cap the user inputs. 

a function should avoid changing the input and should return an error or null in this case if a user input isn't good.

Still, it's nice you came up with this solution. To avoid the if-forest and be a bit my DRY (don't repeat yourself) you could have written a function for that. `int cap_color(int color);`